packages/blocto-sdk/src/providers/ethereum.ts

```diff
@@ -907,4 +911,20 @@ export default class EthereumProvider
       throw ethErrors.rpc.invalidParams('Empty networkList');
     }
   }
+
+  override on(event: string, listener: (arg: any) => void): void {
+    if (this.existedSDK?.isBlocto)
```

Comment:
I think the condition `if (this.existedSDK?.isBlocto)` is more concise and achieves the same effect as `const existedSDK = (window as any).ethereum; if (existedSDK && existedSDK.isBlocto)`. If there are no issues, I'll replace `const existedSDK = (window as any).ethereum; if (existedSDK && existedSDK.isBlocto)` with it.

Reply:
I replaced all the `const existedSDK = (window as any).ethereum; if (existedSDK && existedSDK.isBlocto)` with `if (this.existedSDK?.isBlocto)` in this commit [refactor: enhance existedSDK check condition](https://github.com/blocto/blocto-sdk/pull/288/commits/44c70c71a73833ca36881ec84981b38deca663a8)